specs-go/oci.go

```diff
@@ -102,3 +101,14 @@ func toOCIDevice(d *DeviceNode) spec.Mount {
 		Options:     d.Permissions,
 	}
 }
+
+func toOCILinuxDevice(d *DeviceNode) spec.LinuxDevice {
```

Comment:
Possibly out of scope: Do we also want to modify the cgroups here?

Reply:
@elezar I think cgroups are up to the runtime to setup

Reply:
In my testing, I had to explicitly modify the `spec.Linux.Resources` appending `LinuxDeviceCgroup` entries as in the following example (where the `Inject` method of the `deviceNode` is responsible for modifying the spec):

```golang
func (d deviceNode) Inject(spec *oci.Spec) error {
	if spec.Linux == nil {
		spec.Linux = &oci.Linux{}
	}
	if spec.Linux.Resources == nil {
		spec.Linux.Resources = &oci.LinuxResources{}
	}

	// TODO: These need to be configurable
	deviceFileMode := os.FileMode(8630)
	deviceUID := uint32(0)
	deviceGID := uint32(0)

	deviceMajor := int64(d.Major)
	deviceMinor := int64(d.Minor)

	ociDevice := oci.LinuxDevice{
		Path:     string(d.Path),
		Type:     "c",
		Major:    deviceMajor,
		Minor:    deviceMinor,
		FileMode: &deviceFileMode,
		UID:      &deviceUID,
		GID:      &deviceGID,
	}
	spec.Linux.Devices = append(spec.Linux.Devices, ociDevice)

	ociDeviceCgroup := oci.LinuxDeviceCgroup{
		Allow:  true,
		Type:   "c",
		Major:  &deviceMajor,
		Minor:  &deviceMinor,
		Access: "rwm",
	}

	spec.Linux.Resources.Devices = append(spec.Linux.Resources.Devices, ociDeviceCgroup)

	return nil
}
```
(I know that there are some pieces that are not quite plumbed through there).

The runtime still sets up the cgroups using these resource specifications.

Reply:
Oh - this seems so obvious now that you bring it up :D !
I think it makes sense to have it here - I agree that this should be up to the CDI code (and not spec) to take care.

I'll leave it up to @bart0sh on whether he wants to make it a followup change or part of this PR though.

Reply:
I'd prefer to make it a follow up change.

Reply:
A follow-up is fine.